enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/mtls/MtlsUtils.java

```diff
@@ -0,0 +1,124 @@
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Common functions for certificates.
+ */
+public class MtlsUtils {
+    private static final Logger log = LogManager.getLogger(MtlsUtils.class);
+    public static boolean isPublicCertificate(String cert) {
+        return cert.contains(APIConstants.BEGIN_CERTIFICATE_STRING) &&
+                cert.contains(APIConstants.END_CERTIFICATE_STRING);
+    }
+
+    public static String getCertContent(String cert) {
+        return getCertContent(cert, false);
+    }
+    public static String getCertContent(String cert, boolean isEncoded) {
+        if (isEncoded) {
+            try {
+                cert = URLDecoder.decode(cert, "UTF-8");
+            } catch (UnsupportedEncodingException e) {
+                log.debug("Provided client certificate is unable to decode.");
+                throw new SecurityException(e);
+            }
+        }
+        if (isPublicCertificate(cert)) {
+            String content = cert.replaceAll(APIConstants.BEGIN_CERTIFICATE_STRING, "")
+                    .replaceAll(APIConstants.END_CERTIFICATE_STRING, "");
+            return content.trim();
+        }
+        log.debug("Provided client certificate is not a public certificate.");
+        return "";
+    }
+
+    public static KeyStore createTrustStore(List<Certificate> clientCertificates) throws KeyStoreException {
+        KeyStore trustStore = KeyStore.getInstance("JKS");
+        try {
+            trustStore.load(null, null);
+        } catch (IOException | NoSuchAlgorithmException | CertificateException e) {
+            log.debug("Creating the client certificate truststore was unsuccessful.");
+            throw new SecurityException(e);
+        }
+        for (Certificate certificate : clientCertificates) {
+            X509Certificate x509cert;
+            String alias = certificate.getAlias();
+            String cert = certificate.getContent().toStringUtf8();
+            String certContent = getCertContent(cert);
+            try {
+                x509cert = getX509Cert(certContent);
+            } catch (CertificateException e) {
+                throw new SecurityException(e);
+            }
+            trustStore.setCertificateEntry(alias, x509cert);
+        }
+        return trustStore;
+    }
+
+    public static X509Certificate getX509Cert(String certContent)
+            throws CertificateException {
+        CertificateFactory cf = CertificateFactory.getInstance("X.509");
+        byte[] bytes = Base64.decodeBase64(certContent);
+        X509Certificate x509Certificate;
+        try (ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(bytes)) {
+            x509Certificate =
+                    (X509Certificate) cf.generateCertificate(byteArrayInputStream);
+        } catch (IOException e) {
+            log.debug("Unable to generate x509 certificate format.");
```

Comment:
This has to be an error log I guess.

Reply:
Or else we can catch it from upper layer and log the error.

Reply:
Converted into an error log. Since this will be called from several places, it is difficult to log from an upper layer.